src/query/expression/src/block.rs

```diff
@@ -104,6 +115,11 @@ impl DataBlock {
         DataBlock::new(columns, num_rows)
     }
 
+    #[inline]
+    pub fn attach_filename(self, filename: String) -> Self {
+        Self::new_with_belong_to(self.columns, self.num_rows, filename)
```

Comment:
`attach_filename` should not used in block.

Reply:
if the block has no relation with the original file, we hard to filter block when doing skipfile.

Reply:
Can we add it to the block meta?